scripts/SANS/sans/gui_logic/models/batch_process_runner.py

```diff
@@ -59,7 +59,11 @@ def _process_states_on_thread(self, row_index_pair, get_states_func, use_optimiz
         for row, index in row_index_pair:
 
             # TODO update the get_states_func to support one per call
-            states, errors = get_states_func(row_entries=[row])
+            try:
+                states, errors = get_states_func(row_entries=[row])
+            except Exception as e:
+                self.row_failed_signal.emit(index, str(e))
+                continue
 
             assert len(states) + len(errors) == 1, \
```

Comment:
We discussed that the assert prevents a failed signal from being overwritten by a success signal, since the there is at most one item being processed in the loop.  This needs maintenance sometime, but not today.